packages/next/src/server/lib/route-resolver.ts

```diff
@@ -246,6 +233,8 @@ export async function makeResolver(
       route.type === 'header' ||
       route.name === 'catchall route' ||
       route.name === 'middleware catchall' ||
+      route.name ===
+        `_next/${CLIENT_STATIC_FILES_PATH}/${buildId}/${DEV_MIDDLEWARE_MANIFEST}` ||
```

Comment:
This allows the router to handle this specific route, returning our `MiddlewareMatcher[]` JSON body.